Applications/Spire/Include/Spire/Spire/RealSpinBoxModel.hpp

```diff
@@ -0,0 +1,126 @@
+#ifndef SPIRE_REAL_SPIN_BOX_MODEL_HPP
+#define SPIRE_REAL_SPIN_BOX_MODEL_HPP
+#include <QLocale>
+#include "Spire/Spire/SpinBoxModel.hpp"
+#include "Spire/Ui/RealSpinBox.hpp"
+
+namespace Spire {
+
+  //! Represents a RealSpinBox's data model.
+  class RealSpinBoxModel {
+    public:
+
+      virtual ~RealSpinBoxModel() {};
+
+      //! Returns an increment value.
+      /*
+        \param modifiers The modifiers to get the increment value of.
+      */
+      virtual RealSpinBox::Real get_increment(
+        Qt::KeyboardModifiers modifiers) const = 0;
+
+      //! Returns the initial value.
+      virtual RealSpinBox::Real get_initial() const = 0;
+
+      //! Returns the minimum acceptable value.
+      virtual RealSpinBox::Real get_minimum() const = 0;
+
+      //! Returns the maximum acceptable value.
+      virtual RealSpinBox::Real get_maximum() const = 0;
+  };
+
+  //! Wraps a SpinBoxModel to provide a generic interface.
```

Comment:
A class that transforms one interface into another is known as an ```Adaptor```. Rename this from ```RealSpinBoxModelImpl``` to ```RealSpinBoxAdaptorModel```.